patches/minecraft/net/minecraft/item/ItemBow.java.patch

```diff
@@ -21,7 +21,7 @@
                  if ((double)f >= 0.1D)
                  {
 -                    boolean flag1 = flag && itemstack.func_77973_b() == Items.field_151032_g;
-+                    boolean flag1 = entityplayer.field_71075_bZ.field_75098_d || (itemstack.func_77973_b() instanceof ItemArrow ? ((ItemArrow)itemstack.func_77973_b()).isInfinite(itemstack, p_77615_1_, entityplayer) : false);
++                    boolean flag1 = entityplayer.field_71075_bZ.field_75098_d || itemstack.func_77973_b() instanceof ItemArrow && ((ItemArrow) itemstack.func_77973_b()).isInfinite(itemstack, p_77615_1_, entityplayer);
```

Comment:
Can you keep the parenthesis so that it's `a || (b && c)` instead of `a || b && c`?
Your operator precedence is correct but it can easily confuse people.